core/src/main/java/com/linecorp/armeria/internal/UuidUtil.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal;
+
+import java.util.UUID;
+import java.util.function.Supplier;
+
+import io.netty.util.internal.ThreadLocalRandom;
+
+public final class UuidUtil {
+
+    private static final byte[] HEXDIGITS = { '0', '1', '2', '3', '4', '5', '6', '7',
+                                              '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
+
+    /**
+     * Returns the {@link Supplier} that generates a random {@link UUID}. Similar to {@link UUID#randomUUID()}
+     * except that it uses a cryptographically insecure random for higher performance.
+     */
+    public static Supplier<UUID> randomGenerator() {
+        return RandomUuidGenerator.INSTANCE;
+    }
+
+    /**
+     * Returns a newly generated random {@link UUID}.
+     */
+    public static UUID random() {
```

Comment:
This PR made me realize, we should probably use `String` as the type of the ID rather than `UUID` itself - no reason to force the user to specify a `UUID` if they'd prefer to specify some other type of `String`.

Reply:
That's a good idea. Let me send a follow-up PR for that once this is merged.

Reply:
A follow up PR is ok but wouldn't it affect most of this change? Even abbrev becomes much simpler I think :)

Reply:
That's a good point. :sweat_smile: What do you think about defining an interface for a request ID?

```java
interface RequestId {
    String asText();
    long asLong(); // Could be used as a Brave Span ID?
    String abbreviate();
}
```

Reply:
@adriancole We are going to add an `id` property to `RequestContext`, which will be a random 64-bit integer by default, which is identical to Brave's span ID. I think it will be great if a user can use the same value for `RequestContext.id` and Span ID for easier correlation. However, it seems like span ID generation is currently hard-coded in `Tracer.nextId()`. What do you think about allowing the customization of span ID generation?

Reply:
We've actually been talking about allowing customization of more Tracer / Tracing defaults when creating a span - this would fit well in that model. The biggest downside would be a bad function (e.g. not actually random) breaking tracing - unlike breaking logs, breaking tracing would affect other services/teams.

So given the potential danger, I would lean against customizing span ID since the benefit doesn't seem all that huge (the log integration already makes sure span ids are logged, they happen to be different from the Armeria ID but I think it's fairly expected). I'm also wary of tying the lifecycle of this uid to the lifecycle of distributed tracing (off the top of my head I don't know if our uid currently has the same behavior for retrying client for example and am not sure if it should or not).

FWIW, my plan for this customizer was to always return empty string since I always apply brave, which is a solution I'm happy with. Might be a little, but not that much happier to have a shared ID.